pyproject.toml

```diff
@@ -22,12 +22,13 @@ pyyaml = "6.0"
 requests = "2.28.1"
 keyring = "24.2.0"
 python-box = "6.0.0"
-pydantic = "2.3.0"
-huggingface-hub = "0.16.4"
+pydantic = ">=2.3.0"
+huggingface-hub = ">=0.16.4"
 cloudpickle = "2.2.1"
 networkx = "2.8.4"
 scipy = "1.9.3"
 matplotlib = "3.5.2"
+portalocker = "^2.7.0"
```

Comment:
`portalocker` seems convenient for this use case and is acceptable as a new dependency I think (it is self-contained and does not depends on anything).

Reply:
I agree